### DISCUSSION TITLE
[just for fun]: since we have a client, a server is needed~

### DISCUSSION THREAD

**Discussion by tskdsb** [iceiix/stevenarella#136] [General]
we all know that the java-server is so inefficient:
- low TPS
- high memory use
- awful API extension

we can use rust to fix above, with below targets:
- smaller block for fine structure
- low CPU and memory usage
- friendly API extension for developer and content creator

focus the basic like
- efficient
- extension

then:
- item system
- creature behavior
- ...

more mobs, blocks, items, skills, effects, models will produce by out player with the friendly API extension...

what a beautiful world ！

**Comment by iceiix** [comment]
Alternative server list: https://wiki.vg/Server_List - some are written in Rust (in part or all)

**Comment by iceiix** [comment]
An interesting new server implementation written in Rust: https://github.com/feather-rs/feather/
(mentioned in https://github.com/iceiix/stevenarella/issues/167#issuecomment-529775843)

**Comment by Titaniumtown** [comment]
There's also cuberite (https://github.com/cuberite/cuberite)

**Reply by PureTryOut** [reply]
Cuberite definitely is the most promising alternative option in my eyes. Some skilled devs work on it, there is a great plugin API, and there is a lot of stuff implemented. Still a long way to go till it's fully usable for survival servers though.

https://github.com/cuberite/cuberite/issues/4888

**Comment by iceiix** [comment]
I think Cuberite is among the most promising alternative server implementations (though it is C++/Lua, not Rust). feather-rs also looks to be coming a long way. Could be interesting to test this client with these server implementations.

**Comment by Titaniumtown** [comment]
indeed. I know that cuberite can use like 90% less resources than vanilla, or even paper. I've never looked into feather-rs before.

**Comment by tskdsb** [comment]
thanks, that's enough

/close

**Comment by tristan-f-r** [comment]
https://github.com/Minestom/Minestom minestom would also be a good option -- lightweight and extendable
